### PR TITLE
Add tumblr plugin

### PR DESCRIPTION
**Pull request by mveytsman**
I added a plugin to post lolcommits to a Tumblr blog

### CODE REVIEW

**Comment by mveytsman** [comment]
It looks like my tests are failing due to RuboCop errors in files in `lib/lolcommits/capturer/` which I didn't touch.

I assume they were there before?

**Comment by matthutchinson** [comment]
I'll take a look at any rubocop fails in master now ...

**Comment by matthutchinson** [comment]
Apologies, I've fixed those up rubocop fails in master.  Rebase or merge latest and you should be good to go.

**Comment by mveytsman** [comment]
Thanks, done!

**Comment by mveytsman** [comment]
Hey @matthutchinson  are you good to merge this?

Thanks!

**Comment by matthutchinson** [comment]
Apologies, i'd forgotten about this one, i'll try out your branch tomorrow and if all is OK, i'll merge it in and release a new gem version shortly after. Thanks for your hard work on the plugin!

File: lib/lolcommits/plugins/lol_tumblr.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+module Lolcommits
+  class LolTumblr < Plugin
+    TUMBLR_API_ENDPOINT    = 'https://www.tumblr.com'
+    TUMBLR_CONSUMER_KEY    = '2FtMEDpEPkxjoUdkpHh42h9wqTu9IVS7Ra0QyNZGixdCvhllN2'
+    TUMBLR_CONSUMER_SECRET = 'qWuvxgFUR2YyWKtbWOkDTMAiBEbj7ZGaNLaNQPba0PI1N4JpBs'
+
+    def run_postcapture
+      return unless valid_configuration?
+      puts 'Posting to Tumblr'
+      r = client.photo(configuration['tumblr_name'], :data => runner.main_image)
```
Comment (matthutchinson) [inline lib/lolcommits/plugins/lol_tumblr.rb]: I assume `r` is a tumblr response here and you're checking for a key to ensure it uploaded OK. Can I suggest you rename this to `response` and is it possible this upload could fail here? if so perhaps rescue is needed to cover that case, and show the post FAILED message..

**Comment by matthutchinson** [comment]
All good, tested and is working well.  I have one minor comment (see above).  Its not a huge blocker, but if you could handle the potential for errors here, I think we're good to merge :+1:

File: lib/lolcommits/plugins/lol_tumblr.rb (hunk 1 above)
Comment (matthutchinson) [inline lib/lolcommits/plugins/lol_tumblr.rb]: I've checked the `tumblr_client` gem and it uses Faraday, so rescuing on a `Faraday::Error` here should be enough. I'll merge this PR now and add the relevant rescue myself.

**Comment by matthutchinson** [comment]
Merged!  Thanks for the plugin work!

**Comment by mveytsman** [comment]
Hey @matthutchinson I just wanted to apologize that I missed your comments on this before.

Thanks for doing the changes and merging it!

**Comment by matthutchinson** [comment]
No worries, thanks for contributing :+1:

**Comment by matthutchinson** [comment]
Hi @mveytsman 👋

So you may (or may not) have noticed that we [recently extracted](https://github.com/mroth/lolcommits/issues?utf8=✓&q=label%3Aarchitecture+label%3Aplugin+) all lolcommit plugins to external gems!

🏅 Since you contributed this tumblr plugin, I've added you (as an admin) to the new [lolcommits-tumblr](https://github.com/lolcommits/lolcommits-tumblr) repo.

Of course **you are under no obligation** to maintain this code or gem going forward. This is just a quick message to explain what has happened and give you the opportunity to take ownership again.

If you're interested in picking this up, **send me your email address** (to matt/at/hiddeloop.com) and I'll add you as a new gem owner (with publishing rights) [here](https://rubygems.org/gems/lolcommits-tumblr).

Otherwise I will continue to maintain the gem, and make sure things are kept up to date.

---

To get an idea of how gem plugins work, check out the `sample_plugin` [README](https://github.com/lolcommits/lolcommits-sample_plugin) / [RubyDocs](http://www.rubydoc.info/github/lolcommits/lolcommits-sample_plugin) or browse through [other plugin](https://github.com/lolcommits) code.

The main reasons for extraction were to:

* remove the ever-growing gem dependency list in the main lolcommits gem
* start to unify configuration code for plugins (common helpers, approaches to OAuth etc.)
* make lolcommit plugin development easier
* add more comprehensive plugin tests

That's it, any questions just let me know.

Happy coding!

Matt

**Pull request merged**